### ISSUE TITLE
Out of memory error when running keter on server

### ISSUE THREAD

**Issue by maheshkronecker** [snoyberg/keter#288]
Hello. 

When I upload a keter bundle to `/opt/keter/incoming` and run

```bash
sudo /opt/keter/bin/keter /opt/keter/etc/keter-config.yaml
```

I get the following message :

`process stopped` (after 15 seconds, more if I create a swap file).

I am a bit baffled, as everything was working well and my site was up and running with the same configuration, until it simply stopped working.

```bash
grep -i kill /var/log/*

/var/log/syslog:2024-03-07T10:46:16.278207+01:00 debian kernel: [ 5841.835003] ghc_ticker invoked oom-killer: gfp_mask=0x140cca(GFP_HIGHUSER_MOVABLE|__GFP_COMP), order=0, oom_score_adj=0
/var/log/syslog:2024-03-07T10:46:16.278886+01:00 debian kernel: [ 5841.835152]  oom_kill_process.cold+0xb/0x10
/var/log/syslog:2024-03-07T10:46:16.279094+01:00 debian kernel: [ 5841.835531] oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=/,mems_allowed=0,global_oom,task_memcg=/user.slice/user-1001.slice/session-10.scope,task=keter,pid=1796,uid=0
/var/log/syslog:2024-03-07T10:46:16.279095+01:00 debian kernel: [ 5841.835567] Out of memory: Killed process 1796 (keter) total-vm:1074014088kB, anon-rss:3734684kB, file-rss:0kB, shmem-rss:0kB, UID:0 pgtables:7444kB oom_score_adj:0
/var/log/syslog:2024-03-07T10:46:16.305854+01:00 debian systemd[1]: session-10.scope: A process of this unit has been killed by the OOM killer.
```

```
cat etc/keter-config.yaml
```

```yaml
root: ..

rotate-logs: true

listeners:
    - host: "*4" # Listen on all IPv4 hosts

    - host: "*4"
      key: /etc/appa/privkey.pem
      certificate: /etc/appa/fullchain.pem
      session: true
```

The error persists if I remove the https listener in the yaml file.

**Comment by maheshkronecker** [comment]
I figured something out : it works again it I set `rotate-logs` to false. I guess the `keter.log` file is getting too big. 

```bash
ls -l /opt/keter/log/
total 5172
-rw-r--r-- 1 root root   40534  7 mars  08:52 app-mcqa.log
-rw-r--r-- 1 root root 5244238  7 mars  11:34 keter.log
```

Once the file `keter.log` is deleted and `rotate-logs` set back to `true`, it still works.

I was expecting keter to take care of this. Isn't it the purpose of the  `rotate-logs` option ?

Thanks for any advice on this.

**Comment by jappeace** [comment]
@RiugaBachi do you perhaps remember why it was set to LogFile instead of LogFileTimedRotate ?
should we put a smaller limit instead of intmax? any reason not to?

https://github.com/snoyberg/keter/blob/28ba9b538b9e3850922c5e59a8e3038425a11f9a/src/Keter/Logger.hs#L41

**Comment by RiugaBachi** [comment]
I think we just decided to rotate based on size and not time back then, since there's no size config for TimedRotate and it could in theory explode in size within the rotation interval. And nah I think int max was somewhat arbitrary and can be changed.

**Comment by jappeace** [comment]
okay, I figured it out. your file is prime for rotation, we recently replaced the custom rotation logic with a library called fastlogger.
It wants to rotate because you're just over the limit of 5MB 
```
5*1024*1024 =5242880
```

However, we conservatively set the amount of files fast logger keeps to the maximum of integer, which is `9223372036854775807`.
So now the fast logger code starts making a list of that size and does string concatenation on it, essentially running out of memory!

![image](https://github.com/snoyberg/keter/assets/3874017/920b823a-c535-4f58-b0e3-2618525207ad)
https://hackage.haskell.org/package/fast-logger-3.2.2/docs/src/System.Log.FastLogger.File.html#rotate

To address this issue, I'll set the limit to 20, and I'll file a bug report at fast logger because this is rather poorly documented.
